src/preferences/effectmanifesttablemodel.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <QAbstractItemDelegate>
+#include <QAbstractTableModel>
+#include <QMap>
+#include <QVariant>
+
+#include "effects/backends/effectmanifest.h"
+
+/// EffectManifestTableModel represents a list of EffectManifests as a table
+/// with a column for the BackendType and a column for the effect name.
+/// EffectManifestTableModel is used by DlgPrefEffects to allow the user to drag
+/// and drop between the lists of visible and hidden effects. It also allows the
+/// user to rearrange the list of visible effects so they can choose which
+/// effects can be loaded by controllers.
+class EffectManifestTableModel : public QAbstractTableModel {
```

Comment:
Selected some random rows on the effects pref page and then clicked cancel. This was printed on the terminal. I tried to open the preferences again, but now it just hangs.
```
Warning [Main]: QAbstractItemModel::endInsertRows:  Invalid index ( 32 , 1 ) in model EffectManifestTableModel(0x55555a3176d0)
Warning [Main]: QAbstractItemModel::endInsertRows:  Invalid index ( 32 , 0 ) in model EffectManifestTableModel(0x55555a3176d0)
```

Reply:
I can reproduce this even without clicking cancel: Open the Effects page in the preferences, click an effect in the available effects table, then click an empty row in that table, then switch to another pref page. As soon as you switch back to the effects page, mixxx hangs with 100% CPU usage on one core and needs to be killed.

Reply:
I can't reproduce this because I don't see any empty table rows. Can you still reproduce this since I committed 1a19538?